planning/behavior_path_planner/src/utils/utils.cpp

```diff
@@ -1573,60 +1597,99 @@ std::vector<geometry_msgs::msg::Point> calcBound(
 
   std::optional<lanelet::Polygon3d> current_polygon{std::nullopt};
   std::vector<size_t> current_polygon_border_indices;
-  for (size_t bound_point_idx = 0; bound_point_idx < bound_points.size(); ++bound_point_idx) {
-    const auto & bound_point = bound_points.at(bound_point_idx);
-    const auto polygon = getPolygonByPoint(route_handler, bound_point, "hatched_road_markings");
-
-    bool will_close_polygon{false};
-    if (!current_polygon) {
-      if (!polygon) {
-        output_points.push_back(lanelet::utils::conversion::toGeomMsgPt(bound_point));
-      } else {
-        // There is a new additional polygon to expand
-        current_polygon = polygon;
-        current_polygon_border_indices.push_back(
-          get_corresponding_polygon_index(*current_polygon, bound_point.id()));
+  for (const auto & drivable_lane : drivable_lanes) {
+    // extract target lane and bound.
+    const auto bound_lane = is_left ? drivable_lane.left_lane : drivable_lane.right_lane;
+    const auto bound = is_left ? bound_lane.leftBound3d() : bound_lane.rightBound3d();
+
+    if (bound.size() < 2) {
+      continue;
+    }
+
+    // expand drivable area by intersection areas.
+    const std::string id = bound_lane.attributeOr("intersection_area", "else");
+    if (enable_expanding_intersection_areas && id != "else") {
```

Comment:
```
    const bool isIntersectionArea = enable_expanding_intersection_areas && intersection_area != "else";

    // expand drivable area by intersection areas.
    if (isIntersectionArea) {
```

Reply:
Thanks for your advice! I fixed in https://github.com/autowarefoundation/autoware.universe/pull/4206/commits/1e60bcb87a55dab28ac89e8486de728dd2bc6f6c.